core/unit_test/tools/TestKernelNames.cpp

```diff
@@ -0,0 +1,170 @@
+
+    auto my_lambda_with_tag = KOKKOS_LAMBDA(WorkTag, int){};
+    Kokkos::parallel_for(my_label,
+                         Kokkos::RangePolicy<ExecutionSpace, WorkTag>(0, 1),
+                         my_lambda_with_tag);
+    ASSERT_EQ(last_parallel_for, my_label);
+
+    Kokkos::parallel_for(Kokkos::RangePolicy<ExecutionSpace, WorkTag>(0, 1),
+                         my_lambda_with_tag);
+    ASSERT_EQ(last_parallel_for,
+              std::string(typeid(my_lambda_with_tag).name()) + "/" +
+                  typeid(WorkTag).name());
+  }
+
+  Kokkos::Tools::Experimental::set_begin_parallel_for_callback(nullptr);
+}
+
+void test_kernel_name_parallel_reduce() {
+  Kokkos::Tools::Experimental::set_begin_parallel_reduce_callback(
+      get_parallel_reduce_kernel_name);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+
+  {
+    std::string my_label = "my_parallel_reduce_range_policy";
+    float my_result;
+
+    auto my_lambda = KOKKOS_LAMBDA(int, float&){};
+    Kokkos::parallel_reduce(my_label, Kokkos::RangePolicy<ExecutionSpace>(0, 1),
+                            my_lambda, my_result);
+    ASSERT_EQ(last_parallel_reduce, my_label);
+
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(0, 1),
+                            my_lambda, my_result);
+    ASSERT_NE(last_parallel_reduce.find(typeid(my_lambda).name()),
+              std::string::npos)
+        << last_parallel_reduce << " does not contain "
+        << typeid(my_lambda)
+               .name();  // internally using
+                         // Impl::CombinedFunctorReducer but the name should
+                         // still include the lambda as template parameter
+
+    auto my_lambda_with_tag = KOKKOS_LAMBDA(WorkTag, int, float&){};
+    Kokkos::parallel_reduce(my_label,
+                            Kokkos::RangePolicy<ExecutionSpace, WorkTag>(0, 1),
+                            my_lambda_with_tag, my_result);
+    ASSERT_EQ(last_parallel_reduce, my_label);
+
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace, WorkTag>(0, 1),
+                            my_lambda_with_tag, my_result);
+    ASSERT_NE(last_parallel_reduce.find(typeid(my_lambda_with_tag).name()),
+              std::string::npos)
+        << last_parallel_reduce << " does not contain "
+        << typeid(my_lambda_with_tag).name();  // ditto and below check that it
+                                               // ends with the WorkTag
+    auto suffix = std::string("/") + typeid(WorkTag).name();
+    ASSERT_EQ(last_parallel_reduce.find(suffix),
+              last_parallel_reduce.length() - suffix.length());
+  }
+
+  Kokkos::Tools::Experimental::set_begin_parallel_reduce_callback(nullptr);
+}
+
+void test_kernel_name_parallel_scan() {
+  Kokkos::Tools::Experimental::set_begin_parallel_scan_callback(
+      get_parallel_scan_kernel_name);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+
+  {
+    std::string my_label = "my_parallel_scan_range_policy";
```

Comment:
```suggestion
    const std::string my_label = "my_parallel_scan_range_policy";
```
(similar occurences elsewhere)